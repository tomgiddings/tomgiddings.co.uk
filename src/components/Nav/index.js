// @flow
import React, { useState } from 'react';
import { NavLink } from 'react-router-dom';
import anime from 'animejs';
import { ScrollCheck } from '../Helpers';


const Nav = ({ }) => {
  const [showMenu, setShowMenu] = useState(false)

  const toggleMenu = () => {
    setShowMenu(!showMenu)
  }
  const hideLogo = ScrollCheck();

  return (
    <span>
      <header className="top-bar">
        <span className={ 'logo ' + ( hideLogo ? 'show' : 'hide') }><a href="/">Tom Giddings</a></span>
        <button onClick={toggleMenu} className="toggle-menu"><span className={ 'toggle-menu-inner ' + ( showMenu ? 'show' : 'hide') }></span></button>
      </header>
      <nav className={ 'main-nav ' + ( showMenu ? 'show' : 'hide') }>
        <NavLink className="main-nav-link" exact to="/" onClick={toggleMenu}>Home</NavLink>
        <NavLink className="main-nav-link" exact to="/development" onClick={toggleMenu}>Web Developer</NavLink>
        <NavLink className="main-nav-link" exact to="/broadcast" onClick={toggleMenu}>Broadcast Engineer</NavLink>
        <NavLink className="main-nav-link" exact to="/contact" onClick={toggleMenu}>Contact Me</NavLink>
      </nav>
    </span>
  )
}
export default Nav
