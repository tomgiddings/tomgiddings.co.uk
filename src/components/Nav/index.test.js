import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Nav from './';
import { BrowserRouter } from 'react-router-dom';

afterEach(cleanup);

it('Renders matching snapshot', () => {
  const { asFragment } = render(<BrowserRouter><Nav /></BrowserRouter>);
  expect(asFragment()).toMatchSnapshot();
});
