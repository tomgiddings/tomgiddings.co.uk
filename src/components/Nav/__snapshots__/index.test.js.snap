// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders matching snapshot 1`] = `
<DocumentFragment>
  <span>
    <header
      class="top-bar"
    >
      <span
        class="logo show"
      >
        <a
          href="/"
        >
          Tom Giddings
        </a>
      </span>
      <button
        class="toggle-menu"
      >
        <span
          class="toggle-menu-inner hide"
        />
      </button>
    </header>
    <nav
      class="main-nav hide"
    >
      <a
        aria-current="page"
        class="main-nav-link active"
        href="/"
      >
        Home
      </a>
      <a
        class="main-nav-link"
        href="/development"
      >
        Web Developer
      </a>
      <a
        class="main-nav-link"
        href="/broadcast"
      >
        Broadcast Engineer
      </a>
      <a
        class="main-nav-link"
        href="/contact"
      >
        Contact Me
      </a>
    </nav>
  </span>
</DocumentFragment>
`;
